### DISCUSSION TITLE
Add Webhook for every device

### DISCUSSION THREAD

**Discussion by weebffkorkp22** [seriousm4x/UpSnap#369] [Feature requests]
I would like to propose the implementation of a new feature that enables the automatic generation of webhooks for triggering open and close events.

**Comment by seriousm4x** [comment]
To make sure I understand you correctly:
When a device state changes, upsnap should trigger event x (like a web request to somewhere else)?

**Comment by weebffkorkp22** [comment]
> To make sure I understand you correctly: When a device state changes, upsnap should trigger event x (like a web request to somewhere else)?

Thank you for your response. 

The objective is to enable external devices to initiate a request that will, in turn, trigger the opening or closing of a device.

**Comment by seriousm4x** [comment]
So you mean something like a rest api? There already is one: https://github.com/seriousm4x/UpSnap/wiki/Rest-API
You can send a web request from any device to pwoer/shutdown any device. Does that resolve the issue?
